### ISSUE TITLE
Support for Mohist on 1.12

### ISSUE THREAD

**Issue by vico93** [bergerhealer/BKCommonLib#91]
Hi!

Sorry for this request, i know 1.12 is kinda a dead version for developement, but i think its worth a shot.

I tried to load BKCommonLib on [Mohist](https://github.com/Mohist-Community/Mohist), the only active Paper/Forge hybrid at the moment, but i got [this crash](https://paste.dimdev.org/ejubalegaw.md) and neither BKCL or TrainCarts are activated.

I posted [the issue](https://github.com/Mohist-Community/Mohist/issues/191) on their repo but @Mgazul said:

> Not compatible with BKCommonLib, there are too many problems

There's any thing to improve compatibility which can be done on BKCL's side? Thanks in advance!

**Comment by bbayu123** [comment]
I believe we can't do anything about it, adding compatibility for it would require a complete remap of the NMS calls, which comprise a large amount of the plugin. In layman's terms, the whole plugin will need to be rewritten.

**Comment by bergerkiller** [comment]
Yep, this is just one of many forge forks. It has supported forge in the very distant past, so not all hope is lost. The problem is the absurd number of forge forks out there, that it becomes a lot of effort to support one that very little people use.

**Comment by bergerkiller** [comment]
To add to this: the main thing I need right now is access to the remapper. This is a component that server implementations use to remap net.minecraft.server classes, method names and field names from/to CraftBukkit <> Forge. Without this I can do nothing.

MCPC+ in the distant past had an excellent remapper, which I could hook into. Every single implementation, Mohist included, has such a one. But each implementation, for reasons beyond me, uses a different one. Adding support for just one takes a very long time.

**Comment by bergerkiller** [comment]
@vico93 I've opened a poll on our (Traincarts) dev discord server in the forge-support channel to add support. I put in a vote for Mohist on your behalf.

**Comment by vico93** [comment]
Thanks

**Comment by bergerkiller** [comment]
So it's taken some months, but there is now rudimentary support for Mohist (and Magma/CatServer) in BKCommonLib. Feel free to check it out, and report back any issues you may find. There are some warnings at startup which can be ignored.

**Comment by vico93** [comment]
Oh thank you so much! Will test it right now!

Also i will close both the other issues pertinent to this one!

Thanks!
